### DISCUSSION TITLE
how to get shared area between grains in a periodic tess?

### DISCUSSION THREAD

**Discussion by nishant-mistry** [neperfepx/neper#491] [Q&A] [answered]
I am trying to generate a symmetric table with the areas shared between neighboring grains (considering a periodic structure). I tried 
```
neper -T -n 100 -periodicity all -format tess -statcell id,vol -statpoly id,npolynb,npolys -statface id,per,area,polys
```
however, the .stpoly file gives me poly ids which are more than number of grains, 100 in this case, i.e.,
```
1 18 13 21 23 39 54 55 58 62 68 69 74 75 96 1103 1104 1149 1157 1177
2 13 35 43 48 56 63 71 516 518 527 533 554 583 592
3 18 4 41 49 61 70 76 77 93 1317 1324 1536 1579 1589 1601 1613 1623 1639 1675
4 15 3 5 15 49 57 77 86 91 93 98 1601 1623 1654 1668 1674
5 13 4 11 38 41 51 57 77 86 90 91 98 100 2265
6 14 14 25 28 33 44 84 94 1407 2208 2235 2256 2271 2273 2297
```

I guess the IDs > 100 would be from the imaginary periodic seed around the initial 100 grains. In what manner are these ID-offsets assigned? 

Also, the .stface file gives something like this:
```
1 0 0.040999343710 1 23
2 1 0.036094438252 1
3 1 0.003681311216 1
4 1 0.002812604687 1
5 0 0.033939132953 1 68
6 0 0.027063562664 1 58
7 0 0.050666123552 1 74
8 0 0.005925348018 1 62
9 0 0.000492351538 1 69
10 1 0.036006647281 1
11 0 0.020183389220 1 75
12 0 0.001212726218 1 54
```
leaving the imaginary periodic seed id blank.

**Comment by rquey** [comment, marked as answer]
I've modified `-statpoly npolys` to provide the actual neighbors (considering periodicity) instead of numbers larger than the total number of polys (the previous values can still be obtained using `nseeds`, if it may be useful...).  I've also added `-statpoly npolys_unsort`, which does not sort the ids (something you might be interested in).

I've fixed `-statface polys`, which now provides the 2 polys.
